src/expandable-section/expandable-section-header.tsx

```diff
@@ -131,26 +131,26 @@ const ExpandableHeaderTextWrapper = ({
   onKeyDown,
 }: ExpandableHeaderTextWrapperProps) => {
   const isContainer = variant === 'container';
-  const isDefault = variant === 'default';
   const HeadingTag = headingTagOverride || 'div';
   const hasInteractiveElements = isContainer && (headerInfo || headerActions);
   const listeners = { onClick, onKeyDown, onKeyUp };
 
-  // If interactive elements are present, constrain the clickable area to only the icon and the header text
-  // to prevent nesting interactive elements.
-  const headerButtonListeners = hasInteractiveElements ? listeners : undefined;
-  // For the default variant, include also the immediate wrapper around it to include the entire row
-  // for backwards compatibility, but exclude the description below.
-  const headingTagListeners = !headerButtonListeners && isDefault ? listeners : undefined;
-  // For all other cases, make the entire header clickable for backwards compatibility.
-  const wrapperListeners = !headerButtonListeners && !headingTagListeners ? listeners : undefined;
-
   const description = variantSupportsDescription(variant) && headerDescription && (
     <span id={descriptionId} className={styles[`description-${variant}`]}>
       {headerDescription}
     </span>
   );
 
+  // If interactive elements are present, constrain the clickable area to only the icon and the header text
+  // to prevent nesting interactive elements.
+  const headerButtonListeners = hasInteractiveElements ? listeners : undefined;
+  // For the default and footer variants with description,
+  // include also the immediate wrapper around it to include the entire row for backwards compatibility,
+  // but exclude the description.
+  const headingTagListeners = !headerButtonListeners && !isContainer && description ? listeners : undefined;
```

Comment:
Moving the listeners from the outer wrapper to the inner HeadingTag for the footer variant would break the Side navigation component which uses the footer variant Expandable section internally, because of its test-util selector. Taking a small conservative step back and move the listeners only in the case that a description is present.

Just in case, revamped tests (see above) to make sure that the now recommended `findExpandButton` test-util function works in all expected combinations.